server/game/timeLimit.js

```diff
@@ -30,10 +30,28 @@ class TimeLimit {
         }
     }
 
+    togglePause() {
+        //pause
+        if(this.timeLimitStarted) {
+            this.timeLimitStarted = false;
+            clearInterval(this.timer);
+            this.timer = undefined;
+            let differenceBetweenStartOfTimerAndNow = moment.duration(moment().diff(this.timeLimitStartedAt));
+            this.timeLimitInSeconds -= Math.floor(differenceBetweenStartOfTimerAndNow.asSeconds());
+        } else {
+        //unpause
+            this.timeLimitStarted = true;
+            this.timeLimitStartedAt = new Date();
+            this.timer = setInterval(() => {
+                this.checkForTimeLimitReached();
+            }, 1000);

```

Comment:
For this else, any reason we can't call `startTimer` instead? It seems to be repeating the logic exactly.

Reply:
good catch, totally missed that. fixed!